taglib/mpeg/id3v2/id3v2framefactory.cpp

```diff
@@ -340,7 +340,7 @@ void FrameFactory::rebuildAggregateFrames(ID3v2::Tag *tag) const
   {
     TextIdentificationFrame *tdrc =
       dynamic_cast<TextIdentificationFrame *>(tag->frameList("TDRC").front());
-    UnknownFrame *tdat = static_cast<UnknownFrame *>(tag->frameList("TDAT").front());
+    UnknownFrame *tdat = dynamic_cast<UnknownFrame *>(tag->frameList("TDAT").front());
```

Comment:
With the `dynamic_cast` `tdat` might become `nullptr`, therefore it should be checked. I would change the next line to `if(tdrc && tdat &&`.